src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9003,16 +8994,21 @@ static bool isCandidateUnique(ref MethodSymbol? method, MethodSymbol candidate)
         }
 
         // This method was adapted from LoweredDynamicOperationFactory.GetDelegateType().
-        internal NamedTypeSymbol? GetMethodGroupOrLambdaDelegateType(
-            SyntaxNode syntax,
-            RefKind returnRefKind,
-            TypeWithAnnotations returnType,
-            ImmutableArray<RefKind> parameterRefKinds,
-            ImmutableArray<DeclarationScope> parameterScopes,
-            ImmutableArray<TypeWithAnnotations> parameterTypes,
-            ImmutableArray<ConstantValue?> parameterDefaultValues,
-            bool hasParamsArray)
+        internal NamedTypeSymbol? GetMethodGroupOrLambdaDelegateType(SyntaxNode syntax, MethodSymbol methodSymbol)
         {
+            var parameters = methodSymbol.Parameters;
+            var parameterRefKinds = methodSymbol.ParameterRefKinds;
+            var parameterTypes = methodSymbol.ParameterTypesWithAnnotations;
+            var returnType = methodSymbol.ReturnTypeWithAnnotations;
+            var returnRefKind = methodSymbol.RefKind;
+            var parameterScopes = parameters.Any(p => p.EffectiveScope != DeclarationScope.Unscoped) ?
+                parameters.SelectAsArray(p => p.EffectiveScope) :
+                default;
+            var parameterDefaultValues = parameters.Any(p => p.HasExplicitDefaultValue) ?
+                parameters.SelectAsArray(p => p.ExplicitDefaultConstantValue) :
+                default;
+            var hasParamsArray = !parameters.IsEmpty && parameters[^1].IsParams;
```

Comment:
I extracted the info from `methodSymbol` in one place here instead of using the `methodSymbol` directly in rest of this method, mostly to avoid potential merge conflicts.